thorlcr/activities/csvread/thcsvrslave.cpp

```diff
@@ -104,18 +104,20 @@ class CCsvReadSlaveActivity : public CDiskReadSlaveActivityBase
             CDiskPartHandlerBase::open();
             readFinished = false;
 
+            OwnedIFileIO partFileIO;
             {
                 CriticalBlock block(statsCs);
                 if (compressed)
                 {
-                    iFileIO.setown(createCompressedFileReader(iFile, activity.eexp));
```

Comment:
I don't think this is the correct fix.  You should be creating the reader outside of the critical section, and then having code like the following:

      {
                  CriticalBlock block(statsCs);
                  iFileIO.setown(partFileIO.getClear());
     }